doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+:math:`k = 3` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/full_padding_no_strides_transposed.gif
+
+Strides, transposed
+===================
+
+If a zero-padded convolution has a transpose with padding removed, then a
+convolution with strides greater than 1 has a transpose with strides... smaller
+than 1? In some sense, yes: in order to translate the kernel fractionally, we
+insert zeros between input pixels.
+
+Once again, doing so is inefficient and real-world implementations avoid useless
+multiplications by zero, but conceptually it's how the transpose of a strided
+convolution can be thought of. A convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p = 0`
+
+has a transpose described by
+
+* input size :math:`i' = \left\lfloor \frac{i - k}{s} \right\rfloor + 1`,
+* stretched input size :math:`\tilde{i}' = i' + s (i' - 1)`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = k - 1`
+
+where the input is stretched to size :math:`\tilde{i}'` by adding :math:`s - 1`
+zeros between each pixel.
+
+Here is what the transpose of a convolution looks like for :math:`i = 5`,
+:math:`k = 3` and :math:`s = 2`:
+
+.. figure:: conv_arithmetic_figures/no_padding_strides_transposed.gif
+
+Combining padding and strides, transposed
+-----------------------------------------
+
+Zero-padding and non-unit strides are treated independently to transpose a
+convolution. This means that in the most general case a convolution described by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`
+
+has a transpose described by
+
+* input size :math:`i' = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1`,
+* stretched input size :math:`\tilde{i}' = i' + s (i' - 1)`,
+* kernel size :math:`k' = k`,
+* stride :math:`s' = 1`,
+* padding size :math:`p' = k - p - 1`
+
+where the input is stretched to size :math:`\tilde{i}'` by adding :math:`s - 1`
+zeros between each pixel.
+
+Here is what the transpose of a convolution looks like for :math:`i = 5`,
+:math:`k = 3`, :math:`s = 2` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides_transposed.gif
+
+Sometimes numbers are mean
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you recall, it was pointed out earlier that for :math:`s > 1` and given
+:math:`p` and :math:`k` there were multiple input sizes that lead to the same
+output size. Now is the time where we need to take it into account.
+
+Knowing the values for :math:`i'`, :math:`s`, :math:`k` and :math:`p` is not
+sufficient to define a transposed convolution (unless :math:`s = 1`), as the
+converse convolution over feature maps of sizes
+
+.. math::
+
+    i = s (i - 1) + a + k - 2p, \quad a \in \{0, \ldots, s - 1\}
+
+could have all produced a feature map of size :math:`i'`. This is why in
+addition to :math:`i'` we also have to specify the value of :math:`i`. If
```

Comment:
"the value of `i`" --> "the size `i` that we want to recover / we want to go back to"